examples/http-request-errorbudget.yaml

```diff
@@ -0,0 +1,37 @@
+recordingrule:
+- expr: |
+    sum(label_replace(increase(promhttp_metric_handler_requests_total{namespace="default",job="fooapp"}[30d]), "status_class", "${1}xx", "code", "([0-9])..")) by (status_class)
+  labels:
+    job: fooapp
+    namespace: default
+  record: status_class:promhttp_metric_handler_requests_total:increase30d
+- expr: |
+    (
+      0.001000
+    *
+      sum(status_class:promhttp_metric_handler_requests_total:increase30d)
+    )
+  labels:
+    job: fooapp
+    namespace: default
```

Comment:
The problem with adding the labels here is that the `sum` above would also sum up other `status_class:promhttp_metric_handler_requests_total:increase30d` (from different jobs or namespaces). I'd but the `job` and `namespace` selectors into the selector above, and then `sum without (status_class) (...)` so that you don't even need to add the labels down here.

Reply:
I'm going to take a look at using the `sum without`. Not selecting the proper namespace was definitely a bug that I simply overlooked.

Reply:
The `without` is certainly very flexible as it retains all the other labels, which might include special labels that the user wants to keep.

The separate `labels` section requires, IIUC, that the `selectors` aren't negative or regexp matchers. (See my comment above.) Not sure if that's intended.

I don't want to derail this completely, but perhaps it would be best to give the users freedom for the `selectors` (including regexp and negative matchers) and then have a separate config option to specify the labels to be aggregated away (usually `instance`, but then the user could specify across what dimensions the error budget should be grouped).

To stay more faithful to your current approach, I think you at least have to specify (and ideally validate) that the user has configured just straight-forward equality selectors.